### DISCUSSION TITLE
`getStaticProps` gets called 3 times per request on Vercel?

### DISCUSSION THREAD

**Discussion by msafi** [vercel/vercel#7926] [Help] [answered]
I'm observing a strange behavior when I deploy my Next.js app on Vercel. The behavior is causing a spike in my API and function usages.

This behavior is particular to Vercel. When I build and run a production version of my app on my local machine, I don't observe that behavior.

### What is the behavior?

I have a dynamic route with param name `[celeb]`, at the root like so `mydomain.com/[celeb]`. My expectation is when this route receives a request, the `getStaticProps` associated with the route would get called only once. But I'm observing it get called **3 times** per request.

I should add that I'm using Incremental Static Regeneration. When I switch to Static Site Generation, `getStaticProps` gets called only twice instead of 3 times.

I created a minimal project to reproduce the issue here https://github.com/hollowverse/hollowverse/tree/debug-by-elimination

Would appreciate an explanation of what's going on here!

**Comment by wiiiimm** [comment, marked as answer]
This behaviour is currently expected with ISR, as the revalidation happens in multiple passes, for both the HTML page itself and its associated rehydration content (JSON).

The 3 requests you saw came from:
1. The static content (HTML)
2. The fetch for .json data used by Next.js for prefetching
3. The background ISR revalidation process.

Our Infrastructure Engineers are aware and are currently working to allow both revalidations to happen in one pass.

**Reply by vettloffah** [reply]
Is this currently expected behavior with SvelteKit ISR? I see 2-3 `200` requests for every page loaded (unless their is a `Cache HIT` then I only see one):

![image](https://github.com/vercel/vercel/assets/3166892/2af6847b-0a61-41bd-8075-e6e0064e8a90)

![image](https://github.com/vercel/vercel/assets/3166892/86bbba6d-5b64-438f-819c-4b2502a785cc)
